table/block_based/block_based_table_builder.cc

```diff
@@ -1246,6 +1247,20 @@ void BlockBasedTableBuilder::CompressAndVerifyBlock(
         r->ioptions.clock,
         ShouldReportDetailedTime(r->ioptions.env, r->ioptions.stats));
 
+    *type = r->compression_type;
+#ifndef NDEBUG
+    if (r->compression_type != kNoCompression &&
+        g_hack_mixed_compression_in_block_based_table.LoadRelaxed() > 0U) {
+      // If zstd is in the mix, the compression_name table property needs to be
+      // set to it, for proper handling of context and dictionaries.
+      assert(!ZSTD_Supported() || r->compression_type == kZSTD);
+      auto compressions = GetSupportedCompressions();
```

Comment:
Can we do static auto compressions here?

Reply:
Done